Src/ILGPU/Util/TypeExtensions.cs

```diff
@@ -380,6 +403,80 @@ type switch
                 _ => ArithmeticBasicValueType.None,
             };
 
+        /// <summary>
+        /// Forces the given basic type to 32 bits.
+        /// </summary>
+        /// <param name="type">The source type.</param>
+        /// <returns>The limited arithmetic basic value.</returns>
+        public static ArithmeticBasicValueType ForceTo32Bit(
+            this ArithmeticBasicValueType type) =>
+            type switch
+            {
+                ArithmeticBasicValueType.UInt1 => ArithmeticBasicValueType.UInt32,
+                ArithmeticBasicValueType.UInt8 => ArithmeticBasicValueType.UInt32,
+                ArithmeticBasicValueType.UInt16 => ArithmeticBasicValueType.UInt32,
+                ArithmeticBasicValueType.UInt32 => ArithmeticBasicValueType.UInt32,
+                ArithmeticBasicValueType.UInt64 => ArithmeticBasicValueType.UInt32,
+
+                ArithmeticBasicValueType.Int8 => ArithmeticBasicValueType.Int32,
+                ArithmeticBasicValueType.Int16 => ArithmeticBasicValueType.Int32,
+                ArithmeticBasicValueType.Int32 => ArithmeticBasicValueType.Int32,
+                ArithmeticBasicValueType.Int64 => ArithmeticBasicValueType.Int32,
+
+                ArithmeticBasicValueType.Float16 => ArithmeticBasicValueType.Float32,
+                ArithmeticBasicValueType.Float32 => ArithmeticBasicValueType.Float32,
+                ArithmeticBasicValueType.Float64 => ArithmeticBasicValueType.Float32,
+                _ => ArithmeticBasicValueType.None,
+            };
+
+        /// <summary>
+        /// Limits the given basic type to 32 bits.
+        /// </summary>
+        /// <param name="type">The source type.</param>
+        /// <returns>The limited arithmetic basic value.</returns>
+        public static ArithmeticBasicValueType LimitTo32Bit(
+            this ArithmeticBasicValueType type)
+        {
+            switch (type)
+            {
+                case ArithmeticBasicValueType.Int64:
+                    return ArithmeticBasicValueType.Int32;
+                case ArithmeticBasicValueType.UInt64:
+                    return ArithmeticBasicValueType.UInt32;
+                case ArithmeticBasicValueType.Float64:
+                    return ArithmeticBasicValueType.Float32;
+                default:
+                    return type;
+            }
+        }
+
+        /// <summary>
+        /// Limits the given basic type to 64 bits.
+        /// </summary>
+        /// <param name="type">The source type.</param>
+        /// <returns>The limited arithmetic basic value.</returns>
+        public static ArithmeticBasicValueType LimitTo64Bit(
```

Comment:
QUESTION: Should this be called `ForceTo64Bit`?

Reply:
This makes sense! Let me change that.